pkg/controller/images/images.go

```diff
@@ -103,14 +103,15 @@ func createImageForSelf(req router.Request, app *v1.AppInstance) (*v1.ImageInsta
 				return image, nil
 			}
 		}
-	}
 
-	// remove tag from existing images
-	if err := removeTag(req, app, imageName); err != nil {
-		return nil, err
-	}
+		// If a new image is built with the same tag as the image used for this app,
+		// then we don't want to take the tag from the newly built image.
+		if found, err := checkExistingImages(req, app, imageName); err != nil {
```

Comment:
Nit: imageName seems confusing as it is just a tag in the method input?

Reply:
Tags for Acorn `ImageInstances` is the whole image name. For example, `index.docker.io/thedadams/my-image:v1.0.0` is the Acorn tag.